homeworks/week6/Tam-Tran.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+
+"""
+// Given a string S and a string T, find the minimum window in S which will contain all the characters in T
+// Input: S = "xyyzyzyx" T = "xyz”		Output: "zyx"
+"""
+
+from collections import Counter
+
+def check_window_valid(current_count, expected_count):
+    for i in expected_count:
+        if i not in current_count:
+            return False
+    return True
+    
+# time complexity == 2 time loop through an array (2 pointer) == O(n)
+# space complexity O(n): cause we maintain 2 hash:
```

Comment:
Need to define what is n. My understanding is that n = S.length

So the correct answer should be O(M) where M = T.legnth.